ucp/benchmarks/send_recv.py

```diff
@@ -101,10 +117,9 @@ def client(queue, port, server_address, args):
         xp.cuda.runtime.setDevice(args.client_dev)
         xp.cuda.set_allocator(rmm.rmm_cupy_allocator)
 
-    if args.backend == "ucp-async":
-        client = UCXPyAsyncClient(args, xp, queue, server_address, port)
-    elif args.backend == "ucp-core":
-        client = UCXPyCoreClient(args, xp, queue, server_address, port)
+    client = _get_backend_implementation(args.backend)["client"](
+        args, xp, queue, server_address, port
+    )
```

Comment:
@wence- regarding https://github.com/rapidsai/ucx-py/pull/873/files#r961566376 and https://github.com/rapidsai/ucx-py/pull/873/files#r961566555 , I hope this block makes it clear why we didn't merge the conditions there, we have multiple backends now that have different requirements.